lib/application/api_service.dart

```diff
@@ -0,0 +1,13 @@
+import 'package:aedex/application/session/provider.dart';
+import 'package:archethic_dapp_framework_flutter/archethic_dapp_framework_flutter.dart'
+    as aedappfm;
+import 'package:archethic_lib_dart/archethic_lib_dart.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'api_service.g.dart';
+
+@Riverpod(keepAlive: true)
```

Comment:
Why don't move this provider in aedappfm repo ?

Reply:
It depends on `environmentProvider`, which is unique to `dex`.

Previously, we had two sources of truth for environment management :
   - `aedappfm.EnvironmentUtils`, which was then used by `aedappfm` components
   - the `Dex` `sessionProvider`
   
 Because of this, we could have desynchronization between the two. And in `dex` code, we sometimes used one source of truth or the other, which was not consistent.

The idea is to let Dapps manage environments the way they need : In the Dex,  `environment` is defined by `sessionProvider`, but in another app, it could be defined by another component.